### PR TITLE
nixos/stage-1-systemd: Add LVM2 support (+ test)

### PR DESCRIPTION
**Pull request by dasJ**
###### Description of changes

Add LVM support (lvmraid, thin, and vdo).

Draft because this depends on #167349.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/tasks/lvm.nix [hunk 1]
```diff
@@ -31,8 +35,14 @@ in {
       environment.systemPackages = [ cfg.package ];
       systemd.packages = [ cfg.package ];
 
-      # TODO: update once https://github.com/NixOS/nixpkgs/pull/93006 was merged
+      # XXX update once https://github.com/NixOS/nixpkgs/pull/93006 was merged
```
Comment (lovesegfault) [inline nixos/modules/tasks/lvm.nix:38]: The linked PR is closed, maybe the comment should be removed or adjusted?

File: nixos/modules/tasks/lvm.nix [hunk 2]
```diff
@@ -47,13 +57,15 @@ in {
...
+        systemd.initrdBin = lib.mkIf config.boot.initrd.services.lvm.enable [ pkgs.thin-provisioning-tools ];
+
+        extraUtilsCommands = mkIf (!config.boot.initrd.systemd.enable) ''
           for BIN in ${pkgs.thin-provisioning-tools}/bin/*; do
             copy_bin_and_libs $BIN
           done
         '';
 
-        extraUtilsCommandsTest = ''
+        extraUtilsCommandsTest = mkIf (!config.boot.initrd.systemd.enable) ''
```
Comment (lovesegfault) [inline nixos/modules/tasks/lvm.nix:68]: Would it be possible to just ignore these settings at a global level when `boot.initrd.systemd.enable` is set?

That way we don't need this clause everywhere.

**Review by lovesegfault — APPROVED** [review]
Looks good, left some non-blocking comments.

File: nixos/modules/tasks/lvm.nix (hunk 2 above)
Comment (dasJ) [reply nixos/modules/tasks/lvm.nix:68]: Yeah they are just ignored.
But my idea was to get rid of their uses and assert that the options are not set to make sure people don't set these options in their local configs and being surprised they don't do anything anymore.

Comment (lovesegfault) [reply nixos/modules/tasks/lvm.nix:67]: Good idea!

**Pull request merged**